core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -1008,7 +1014,12 @@ public CompletionStage<Void> removeListenerAsync(Object listener) {
 
    @Override
    public ComponentStatus getStatus() {
-      return status;
+      lifecycleLock.lock();
```

Comment:
Status is volatile, is this really necessary?

Reply:
Is this so a certain method blocks when another is running concurrently? I feel like that should be done in the method instead.

Reply:
Yeah. Specifically, if a stop method is running.